P2PEnabledConference.js

```diff
@@ -0,0 +1,947 @@
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    onRemoteTrackAdded(track) {
+        if (track.isP2P && !this.p2pEstablished) {
+            logger.info(
+                'Trying to add remote P2P track, when not in P2P - IGNORED');
+        } else if (!track.isP2P && this.p2pEstablished) {
+            logger.info(
+                'Trying to add remote JVB track, when in P2P - IGNORED');
+        } else {
+            super.onRemoteTrackAdded(track);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * @override
+     */
+    onTransportInfo(jingleSession, transportInfo) {
+        if (this.p2pJingleSession === jingleSession) {
+            logger.info('Doing set transport-info');
+            this.p2pJingleSession.addIceCandidates(transportInfo);
+        }
+    }
+
+    /**
+     * Manually starts new P2P session (should be used only in the tests).
+     */
+    startPeer2PeerSession() {
+        const peers = this.getParticipants();
+
+        // Start peer to peer session
+
+        if (peers.length === 1) {
+            const peerJid = peers[0].getJid();
+
+            this._startPeer2PeerSession(peerJid);
+        } else {
+            throw new Error(
+                'There must be exactly 1 participant '
+                    + 'to start the P2P session !');
+        }
+    }
+
+    /**
+     * Manually stops the current P2P session (should be used only in the tests)
+     */
+    stopPeer2PeerSession() {
+        this._stopPeer2PeerSession();
+    }
+}
+
+/**
+ * This is a fake {@link ChatRoom} passed to the P2P {@link JingleSessionPC}
+ * in order to capture events emitted on its event emitter (Jingle session uses
+ * chat room's emitter to send events).
+ */
+class FakeChatRoomLayer {
+
+    /**
+     * Creates new <tt>FakeChatRoomLayer</tt>
+     * @param p2pConference parent <tt>P2PEnabledConference</tt> instance
+     */
+    constructor(p2pConference) {
+
+        /**
+         * @type P2PEnabledConference
+         */
+        this.p2pConf = p2pConference;
+
+        /**
+         * See whatever docs are provided in
+         * the {@link ChatRoom#connectionTimes}.
+         * @type {Array}
+         */
+        this.connectionTimes = [];
+
+        /**
+         * Maps options of the original <tt>ChatRoom</tt>
+         */
+        this.options = p2pConference.room.options;
+        if (!this.options) {
+            logger.error('ChatRoom.options are undefined');
+        }
+
+        /**
+         * Partial implementation of the <tt>EventEmitter</tt> used to intercept
+         * events emitted by the P2P {@link JingleSessionPC}
+         * @type {EventEmitter}
+         */
+        this.eventEmitter = this._createEventEmitter();
+    }
+
+    /**
+     * Creates fake event emitter used to intercept some of the XMPP events
+     * emitted by {@link P2PEnabledConference.p2pJingleSession}.
+     * @return {EventEmitter}
+     * @private
+     */
+    _createEventEmitter() {
+        const self = this;
```

Comment:
it is needed, there is object defined below, I don't know how to write it otherwise. I guess arrow will not work here